lib/active_record/virtual_attributes/virtual_delegates.rb

```diff
@@ -271,11 +271,11 @@ def self.select_from_alias(to_ref, col, to_model_col_name, src_model_id)
       def self.select_from_alias_table(to_klass, src_relation)
         to_table = to_klass.arel_table
         # if a self join, alias the second table to a different name
-        if to_table.table_name == src_relation.table_name
+        if to_table.name == src_relation.name
           # use a dup to not modify the primary table in the model
           to_table = to_table.dup
           # use a table alias to not conflict with table name in the primary query
-          to_table.table_alias = "#{to_table.table_name}_sub"
+          to_table.instance_variable_set(:@table_alias, "#{to_table.name}_sub")
```

Comment:
@kbrock do we need to set a table alias in this way?  I didn't look to see if this was yours or Nick's comment.  Clearly, they don't want to let us set it so I concerned this will change in the future.

Reply:
aah - now I remember

rails uses a manager to manage the names of the tables (so you can do a self join and it all works)

I had been meaning to change this for the next revision

If I remember correctly, this is tied to 3 test failures - the only real failures for rails.

They are real faiulres.
Pretty sure we don't want `name` here.

Reply:
I'll leave this WIP... feel free to take this change in part or whole and change it while I'm away.

Reply:
I'm putting together a PR for tests that will verify this change